src/Type/Constant/ConstantIntegerType.php

```diff
@@ -88,6 +93,56 @@ public function generalize(GeneralizePrecision $precision): Type
 		return new IntegerType();
 	}
 
+	public function exponentiate(Type $exponent): Type
+	{
+		if ($exponent instanceof UnionType) {
+			$results = [];
+			foreach ($exponent->getTypes() as $unionType) {
+				$results[] = $this->exponentiate($unionType);
+			}
+			return TypeCombinator::union(...$results);
+		}
+
+		if ($exponent instanceof IntegerRangeType) {
+			$min = null;
+			$max = null;
+			if ($exponent->getMin() !== null) {
+				$min = $this->getValue() ** $exponent->getMin();
+			}
+			if ($exponent->getMax() !== null) {
+				$max = $this->getValue() ** $exponent->getMax();
+			}
+
+			if (!is_float($min) && !is_float($max)) {
+				return IntegerRangeType::fromInterval($min, $max);
+			}
+		} elseif ($exponent instanceof ConstantScalarType) {
+			$exponentValue = $exponent->getValue();
+			if (is_int($exponentValue)) {
+				$min = $this->getValue() ** $exponentValue;
```

Comment:
Memo to me: double check... seems duplicated with the following if (constant-scalar) case